### PR TITLE
Core/Spell: SPELL_EFFECT_PULL_TOWARDS correction

### PR DESCRIPTION
**Pull request by Faq**
**Changes proposed:**
Info provided from xvwyh in #24178 comments:

> You might also notice that that spells's effect is SPELL_EFFECT_PULL_TOWARDS, and as it turns out - yes, it too must be implemented as a clientside knockback for players, just like SPELL_EFFECT_PULL_TOWARDS_DEST.
Additionally I believe that SPELL_EFFECT_PULL_TOWARDS must be pulling units not to caster's position, but next to them, e.g. to the edge of their hitbox. Otherwise, what is the point in them being separate effects? However I cannot prove this yet, so it's up to you if you want to change it or not. In case you do: replace pos.Relocate(m_caster); with m_caster->GetFirstCollisionPosition(pos, m_caster->GetCombatReach(), m_caster->GetRelativeAngle(unitTarget)); or something along those lines.
Just bear in mind that SPELL_EFFECT_PULL_TOWARDS is used in many more cases than SPELL_EFFECT_PULL_TOWARDS_DEST, and I can't vouch for all of them being correct after these changes, because I literally just discovered this.
Still up to you, TC devs, if you want to apply the changes to SPELL_EFFECT_PULL_TOWARDS handler as well or not, at the very least it will improve the behavior of the effect (make it clientside, allowing players to collide with solid surfaces, not fall through the ground, work on moving transports etc), but it still won't be anywhere near blizzlike, because thus far I don't have the slightest clue how is it even supposed to look like. Each spell behaves differently.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
tested with https://woehead.way-of-elendil.fr/?npc=15975 and his **Spider Web**
.go creature id 15975

**Known issues and TODO list:** (add/remove lines as needed)

- [x] more tests needed

### CODE REVIEW

**Comment by Faq** [comment]
I encountered problem with this and not able to compile it. Any help would greatly appreciate.

> 'WorldObject::GetFirstCollisionPosition': function does not take 3 arguments

**Comment by Rushor** [comment]
Position WorldObject::GetFirstCollisionPosition(float dist, float angle)
{
    Position pos = GetPosition();
    MovePositionToFirstCollision(pos, dist, angle);
    return pos;
}

change 
Position const* pos;
    m_caster->GetFirstCollisionPosition(pos, m_caster->GetCombatReach(), m_caster->GetRelativeAngle(unitTarget));

to 

Position pos =m_caster->GetFirstCollisionPosition(pos, m_caster->GetCombatReach(), m_caster->GetRelativeAngle(unitTarget));


somewhere in 2014 that was changed

**Comment by Faq** [comment]
thnx

**Comment by Rushor** [comment]
Why Do you closed it?

**Comment by illfated** [comment]
Looks like the merge has overwritten the changes, leaving nothing changed compared to the TC branch.

**Comment by Faq** [comment]
Git found my curvehands.dll :)
Whats wrhong with appeyor?

**Comment by illfated** [comment]
Looks a bit odd, the page says "Log is empty." and "More recent PR #24251 build has been started." which is sort of a contradiction. Not sure, but if you make another change, maybe it will update itself correctly.

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -4392,12 +4392,15 @@ void Spell::EffectPullTowards(SpellEffIndex effIndex)
     if (!unitTarget)
         return;
 
-    float speedXY = m_spellInfo->Effects[effIndex].MiscValue / 10.0f;
-    float speedZ = damage / 10.0f;
-    Position pos;
-    pos.Relocate(m_caster);
+    Position pos = m_caster->GetFirstCollisionPosition(m_caster->GetCombatReach(), m_caster->GetRelativeAngle(unitTarget));
+
+    // This is a blizzlike mistake: this should be 2D distance according to projectile motion formulas, but Blizzard erroneously used 3D distance
```
Comment (illfated) [inline src/server/game/Spells/SpellEffects.cpp]: Maybe make a small change like adding a period dot ( . ) at the end of the comment (unless you know anything else that could need an update).

**Comment by illfated** [comment]
"Hi Appveyor" -- thanks for the chuckle. 😄

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**